magefiles/tests.go

```diff
@@ -40,6 +42,70 @@ func gotestsum() error {
 	return nil
 }
 
+// Tests is a mage target that runs the tests and generates coverage reports.
```

Comment:
I still do not think this is needed. We should, be able to use:

`mage localdev minimal testsuite` here and get the result we want.

Reply:
Hey, [mage localdev no-build](https://github.com/armadaproject/armada/blob/b498d64ac52a98facee494eaa5720e7f5af5b4ff/magefiles/main.go#L180) also does the kind setup thing, which is not required as per the instructions in [make](https://github.com/armadaproject/armada/blob/b498d64ac52a98facee494eaa5720e7f5af5b4ff/makefile#L399) to run these tests, that's why I have not used that.